File: PDOO/Practica1/Java/Irrgarten/src/irrgarten/GameState.java
package irrgarten;

/**
 * This class is used to represent the state of the game
 */
public class GameState {
    private String labyrinthv; 
    private String players;
    private String monsters;
    private int currentPlayer;
    private boolean winner;
    private String log;

    public GameState(String labyrinthv, String players, String monsters, int currentPlayer, boolean winner, String log) {
        this.labyrinthv = labyrinthv;
        this.players = players;
        this.monsters = monsters;
        this.currentPlayer = currentPlayer;
        this.winner = winner;
        this.log = log;
    }

    public String getLabyrinthv() {
        return labyrinthv;
    }

    public String getPlayers() {
        return players;
    }

    public String getMonsters() {
        return monsters;
    }

    public int getCurrentPlayer() {
        return currentPlayer;
    }

    public boolean getWinner() {
        return winner;
    }

    public String getLog() {
        return log;
    }


}

File: PDOO/Practica1/Ruby/Irrgarten/GameState.rb
class GameState
    def initialize(labyrinthv, players, monsters, current_player, winner, log)
        @labyrinthv = labyrinthv
        @players = players
        @monsters = monsters
        @current_player = current_player
        @winner = winner
        @log = log
    end

    def get_labyrinthv
        return @labyrinthv
    end

    def get_players
        return @players
    end

    